doc/api/stream.md

```diff
@@ -883,83 +1035,109 @@ implement Readable streams in your programs.
   * `read` {Function} Implementation for the [`stream._read()`][stream-_read]
     method.
 
-In classes that extend the Readable class, make sure to call the
-Readable constructor so that the buffering settings can be properly
-initialized.
+For example:
+
+```js
+const Readable = require('stream').Readable;
+
+class MyReadable extends Readable {
+  constructor(options) {
+    // Calls the stream.Readable(options) constructor
+    super(options);
+  }
+}
+```
+
+Or, when using pre-ES6 style constructors:
+
+```js
+const Readable = require('stream').Readable;
+const util = require('util');
+
+function MyReadable(options) {
+  if (!(this instanceof MyReadable))
+    return new MyReadable(options);
+  Readable.call(this, options);
+}
+util.inherits(MyReadable, Readable);
```

Comment:
The notes about `options.write` / `_write` apply to `options.read` / `_read` as well!